src/OpenLoco/src/Objects/ObjectIndex.cpp

```diff
@@ -1058,4 +1058,190 @@ namespace OpenLoco::ObjectManager
+
+        if (!validateObjectTypeSelection(objectFlags, ObjectType::building, atLeastOneSelected))
+        {
+            GameCommands::setErrorText(StringIds::town_building_type_must_be_selected);
+            return ObjectType::building;
+        }
+
+        if (!validateObjectTypeSelection(objectFlags, ObjectType::interfaceSkin, atLeastOneSelected))
+        {
+            GameCommands::setErrorText(StringIds::interface_type_must_be_selected);
+            return ObjectType::interfaceSkin;
+        }
+
+        if (!validateObjectTypeSelection(objectFlags, ObjectType::vehicle, atLeastOneSelected))
+        {
+            GameCommands::setErrorText(StringIds::vehicle_type_must_be_selected);
+            return ObjectType::vehicle;
+        }
+
+        if (!validateObjectTypeSelection(objectFlags, ObjectType::land, atLeastOneSelected))
+        {
+            GameCommands::setErrorText(StringIds::land_type_must_be_selected);
+            return ObjectType::land;
+        }
+
+        if (!validateObjectTypeSelection(objectFlags, ObjectType::currency, atLeastOneSelected))
+        {
+            GameCommands::setErrorText(StringIds::currency_type_must_be_selected);
+            return ObjectType::currency;
+        }
+
+        if (!validateObjectTypeSelection(objectFlags, ObjectType::water, atLeastOneSelected))
+        {
+            GameCommands::setErrorText(StringIds::water_type_must_be_selected);
+            return ObjectType::water;
+        }
+
+        if (!validateObjectTypeSelection(objectFlags, ObjectType::townNames, atLeastOneSelected))
+        {
+            GameCommands::setErrorText(StringIds::town_name_type_must_be_selected);
+            return ObjectType::townNames;
+        }
+
+        if (!validateObjectTypeSelection(objectFlags, ObjectType::levelCrossing, atLeastOneSelected))
+        {
+            GameCommands::setErrorText(StringIds::level_crossing_type_must_be_selected);
+            return ObjectType::levelCrossing;
+        }
+
+        if (!validateObjectTypeSelection(objectFlags, ObjectType::streetLight, atLeastOneSelected))
+        {
+            GameCommands::setErrorText(StringIds::street_light_type_must_be_selected);
+            return ObjectType::streetLight;
+        }
+
+        if (!validateObjectTypeSelection(objectFlags, ObjectType::snow, atLeastOneSelected))
+        {
+            GameCommands::setErrorText(StringIds::snow_type_must_be_selected);
+            return ObjectType::snow;
+        }
+
+        if (!validateObjectTypeSelection(objectFlags, ObjectType::climate, atLeastOneSelected))
+        {
+            GameCommands::setErrorText(StringIds::climate_type_must_be_selected);
+            return ObjectType::climate;
+        }
+
+        if (!validateObjectTypeSelection(objectFlags, ObjectType::hillShapes, atLeastOneSelected))
+        {
+            GameCommands::setErrorText(StringIds::map_generation_type_must_be_selected);
+            return ObjectType::hillShapes;
+        }
```

Comment:
Can we turn this into a (static const) struct of <ObjectType, StringId> pairs, that we could then loop over it instead? That would save a lot of repetitive code, I think.

Reply:
That would  be a good idea.